src/rotary-dialer-element.ts

```diff
@@ -0,0 +1,297 @@
+        }
+        50% {
+          transform: rotate(290deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+
+      @keyframes dialZero {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(320deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+    `;
+  }
+
+  private static matchDigit(
+    slots: SVGPathElement,
+    digit: Digit,
+    direction: Direction = Direction.normal,
+    duration = '1000ms',
+    timingFn = 'ease-out'
+  ) {
+    switch (digit) {
+      case 0:
+        slots?.setAttribute('style', `animation: dialZero ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 1:
+        slots?.setAttribute('style', `animation: dialOne ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 2:
+        slots?.setAttribute('style', `animation: dialTwo ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 3:
+        slots?.setAttribute(
+          'style',
+          `animation: dialThree ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case 4:
+        slots?.setAttribute('style', `animation: dialFour ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 5:
+        slots?.setAttribute('style', `animation: dialFive ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 6:
+        slots?.setAttribute('style', `animation: dialSix ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 7:
+        slots?.setAttribute(
+          'style',
+          `animation: dialSeven ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case 8:
+        slots?.setAttribute(
+          'style',
+          `animation: dialEight ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case 9:
+        slots?.setAttribute('style', `animation: dialNine ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      default:
+        slots?.removeAttribute('style');
+    }
+  }
+
+  private addDialerAnim(digit: Digit) {
+    this.handleRotation(digit);
+  }
+
+  private removeDialerAnim() {
+    const slots = this.shadowRoot?.querySelector('#slots');
+    slots?.classList.remove('dialer-anim');
+    slots?.removeAttribute('style');
+  }
+
+  private up(digit: Digit) {
+    this.dial(digit);
+  }
+
+  /**
+   * Exposed lazy dial by applying dial method with a given argument of number from 0-9
+   * @param digit
+   */
+  public dial(digit: Digit) {
+    this.removeDialerAnim();
+    this.addDialerAnim(digit);
+  }
+
+  private onValueChange(event: KeyboardEvent) {
+    const target = event.target as HTMLInputElement;
+    const digit = parseInt(target.value) as Digit;
+    this.dial(digit);
+    target.value = '';
+  }
+
+  private handleRotation(digit: Digit) {
+    const slots = this.shadowRoot?.querySelector('#slots') as SVGPathElement;
+    slots.onanimationend = () => {
```

Comment:
You can use a lit element event listener instead of manually attaching the event listener every time

Reply:
yes that would be better!